### DISCUSSION TITLE
set_pattern() for different loggers with the same sink

### DISCUSSION THREAD

**Discussion by amirych** [gabime/spdlog#1902] [Q&A] [answered]
I'm a little confused with this:

```
#include <spdlog/sinks/basic_file_sink.h>

int main() {
    auto sharedFileSink = std::make_shared<spdlog::sinks::basic_file_sink_mt>("fileName.txt");

    auto firstLogger = std::make_shared<spdlog::logger>("firstLoggerName", sharedFileSink);
    auto secondLogger = std::make_unique<spdlog::logger>("secondLoggerName", sharedFileSink);

    firstLogger->set_pattern("[1] %v");
    secondLogger->set_pattern("[2] %v");


    firstLogger->info("msg1");
    secondLogger->info("msg2");

    return 0;
}
```

fileName.txt content is:

```
[2] msg1
[2] msg2
```

I supposed format pattern setting in one logger should not change one in other, even if loggers share the same sink.
Am I mistaken?

**Comment by tt4g** [comment, marked as answer]
The formatter is an attribute of the sink, not the logger.

`spdlog::logger::set_pattern` is a short hand to set the formatter for each sink in the logger.
https://github.com/gabime/spdlog/blob/100f30043f33277122e0991c83845a2617172ffd/include/spdlog/logger-inl.h#L82-L105

**Reply by amirych** [reply]
Ok, now I see
